### PR TITLE
Email is mandatory but phone number is optional in the enrolment form

### PR DESCRIPTION
**Pull request by nikomakela**
PT-891. Enrolments does not require a phone number anymore.

Added missing required-attributes to emails and names. Removed required-attributes and validations from phone fields. 

Enrolment's email notifications are now mandatory. The checkbox is auto selected and disabled in UI, so it appears as checked but passive. GraphQL mutation does not allow sms only notifications anymore. It should be noted that API still allows them!

### CODE REVIEW

File: src/domain/enrolment/utils.ts [hunk 1]
```diff
@@ -5,15 +5,10 @@ import {
 } from '../../generated/graphql';
 import { EnrolmentFormFields } from './enrolmentForm/constants';
 
-const getNotificationType = (
-  values: EnrolmentFormFields
-): NotificationType | undefined => {
+const getNotificationType = (values: EnrolmentFormFields): NotificationType => {
```
Comment (Sartonon) [inline src/domain/enrolment/utils.ts:8]:
```
ternary might read nicer in this situation :)

return values.hasEmailNotification && values.hasSmsNotification
    ? NotificationType.EmailSms
    : NotificationType.Email;
```

File: src/domain/enrolment/enrolmentForm/EnrolmentForm.tsx [hunk 2]
```diff
@@ -232,10 +234,14 @@ const EnrolmentForm: React.FC<Props> = ({
               <div className={styles.divider} />
 
               <h2>{t('enrolment:enrolmentForm.titleNotifications')}</h2>
+              {errors.hasEmailNotification && touched.hasEmailNotification && (
+                <ErrorMessage>{t(errors.hasEmailNotification)}</ErrorMessage>
+              )}
```
Comment (Sartonon) [inline src/domain/enrolment/enrolmentForm/EnrolmentForm.tsx:239]: email notifications are always checked and disabled, why is this needed?

File: src/domain/enrolment/utils.ts (hunk 1 above)
Comment (nikomakela) [reply src/domain/enrolment/utils.ts:8]: true that. Because of some reason I wanted to keep it easily recoverable / extendable :)

File: src/domain/enrolment/enrolmentForm/EnrolmentForm.tsx (hunk 2 above)
Comment (nikomakela) [reply src/domain/enrolment/enrolmentForm/EnrolmentForm.tsx:239]: Maybe it is not. I didn't think about it much. I thought there could be a way to achieve it some how...

I will take it off.

File: src/domain/enrolment/utils.ts (hunk 1 above)
Comment (Sartonon) [reply src/domain/enrolment/utils.ts:8]: Well it wasn't wrong :)

**Review by Sartonon — APPROVED** [review]
